### PR TITLE
Bob develop

### PR DESCRIPTION
**Pull request by bobverity**
Extension to allow different init values over chains

### CODE REVIEW

File: R/main.R [hunk 1]
```diff
@@ -11,6 +11,40 @@ check_drjacoby_loaded <- function() {
...
+#' @title Define parameters dataframe
+#'
+#' @description Provides a convenient way of defining parameters in the format
+#'   required by \code{run_mcmc()}.
+#'
+#' @param ... define parameters one at a time, for each one defining four
+#'   elements: name, min, max, init.
+#'
+#' @export
+
```
Comment (pwinskill) [inline R/main.R:27]: Could we mabe add an example here - its not clear to me how this is used to define multiple params (can we chain mutiple calls to this function togther somehow?)

File: R/main.R [hunk 2]
```diff
@@ -45,10 +81,12 @@ check_drjacoby_loaded <- function() {
 #' @param coupling_on whether to implement Metropolis-coupling over temperature 
 #'   rungs.
 #' @param GTI_pow values in the temperature ladder are raised to this power.
+#'   Provides a convenient way of concentrating rungs towards one end of the
+#'   temperature scale.
 #' @param beta_manual option to manually define temperature ladder. These values
 #'   are raised to the power \code{GTI_pow}, hence you should use \code{GTI_code
-#'   = 1} to fix powers exactly. If \code{NULL} then an equal spacing of length
-#'   \code{rungs} is used between 0 and 1.
+#'   = 1} if you want to fix powers exactly. If \code{NULL} then an equal
```
Comment (pwinskill) [inline R/main.R:91]: Do values of `beta_manual` need to be beyween 0 and 1?
and
Does the length of beta_manual automatically define/override the # of rungs?

File: vignettes/example.Rmd [hunk 3]
```diff
@@ -50,21 +50,26 @@ sigma_true <- 2
...
 print(df_params)
 ```
 
 In this example we have one parameter (`mu`) that occupies a finite range [-10, 10], and another parameter (`sigma`) that can take any positive value. *drjacoby* deals with different parameter ranges using reparameterisation, which all occurs internally meaning we don't need to worry about these constraints affecting our inference.
 
-Next, we need a likelihood function. This **must** have exactly two input arguments: 1) a vector of parameters, 2) a vector of data, and these **must** be input **in that order**. It also **must** return a single value for the likelihood **in log space**. These constraints on the format of the likelihood function might seem a bit restrictive, but they are needed in order for *drjacoby* to know how to use the function internally. The issue of taking logs is particularly important, as the MCMC will still run even if we forget to take logs, but the results produced will be nonsense!
+Next, we need a likelihood function. This **must** have the following four input arguments, in this order:
+
+1. `params` - a named vector of parameter values
+2. `param_i` - an index for the current parameter being updated
```
Comment (pwinskill) [inline vignettes/example.Rmd:68]: I've missed a lot of iteration here that I assume went into Nick's work - so apologies if my understanding is superficial. I was wondering whether it was possible to wrap `param_i` into a named element of `misc`, seeing as I guess for most users situations it won't be required?

**Review by pwinskill — CHANGES_REQUESTED** [review]
Hey Bob - looks good. Mostly minor comments just asking for a little more info on some of the new functionality.

GHA editing looks fine

File: R/main.R (hunk 1 above)
Comment (bobverity) [reply R/main.R:27]: You're quite right, it was lazy of me not to make an example. Will add one now

File: R/main.R [hunk 4]
```diff
@@ -11,6 +11,40 @@ check_drjacoby_loaded <- function() {
...
+#'
+#' @description Provides a convenient way of defining parameters in the format
+#'   required by \code{run_mcmc()}.
+#'
+#' @param ... define parameters one at a time, for each one defining four
+#'   elements: name, min, max, init.
+#'
+#' @export
+
+define_params <- function(...) {
```
Comment (bobverity) [inline R/main.R:28]: Added a test to check the output of this function against a basic dataframe produced through base R.

File: R/main.R (hunk 2 above)
Comment (bobverity) [reply R/main.R:91]: Great question. For Metropolis-Coupling there is no requirement that rungs go all the way down to zero, so in the code we currently have no requirement for this. But I'm also trying to set us up for the next DrJ paper, in which I think we implement thermodynamic integration and talk about model comparison. In this case we will need temperatures to go to zero, so we might need some extra checks at that point.

File: R/main.R (hunk 4 above)
Comment (bobverity) [reply R/main.R:28]: In offline conversation decided to drop param_i in later PR and move to block method instead

**Pull request merged**